### PR TITLE
Support claiming of children

### PR DESCRIPTION
**Pull request by knatsakis**

### CODE REVIEW

**Comment by ilyam8** [comment]
@knatsakis just a suggestion to make it more readable (from my point of view)


```yaml
          {{- if and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms }}
          command:
            - sh
            - '-c'
            - >
              exec /usr/sbin/run.sh -W set2 cloud global enabled true -W set2 cloud global "cloud base url" "{{ .Values.child.claiming.url }}" -W "claim -token={{ .Values.child.claiming.token }} -rooms={{ .Values.child.claiming.rooms }} -url={{ .Values.child.claiming.url }}"
          {{- end }}
```

=>

```yaml
          {{- with .Values.child.claiming }}
          {{- if and .enabled .token .rooms }}
          command:
            - sh
            - '-c'
            - >
              exec /usr/sbin/run.sh \
              -W set2 cloud global enabled true \
              -W set2 cloud global "cloud base url" "{{ .url }}" \
              -W "claim -token={{ .token }} -rooms={{ .rooms }} -url={{ .url }}"
          {{- end }}
          {{- end }}
```

**Review by ilyam8 — APPROVED** [review]

**Comment by knatsakis** [comment]
> @knatsakis just a suggestion to make it more readable (from my point of view)
> 
> ```yaml
>           {{- if and .Values.child.claiming.enabled .Values.child.claiming.token .Values.child.claiming.rooms }}
>           command:
>             - sh
>             - '-c'
>             - >
>               exec /usr/sbin/run.sh -W set2 cloud global enabled true -W set2 cloud global "cloud base url" "{{ .Values.child.claiming.url }}" -W "claim -token={{ .Values.child.claiming.token }} -rooms={{ .Values.child.claiming.rooms }} -url={{ .Values.child.claiming.url }}"
>           {{- end }}
> ```
> 
> => (more verbose but less error prone and there is no super long lines)
> 
> ```yaml
>           {{- with .Values.child.claiming }}
>           {{- if and .enabled .token .rooms }}
>           command:
>             - sh
>             - '-c'
>             - >
>               exec /usr/sbin/run.sh \
>               -W set2 cloud global enabled true \
>               -W set2 cloud global "cloud base url" "{{ .url }}" \
>               -W "claim -token={{ .token }} -rooms={{ .rooms }} -url={{ .url }}"
>           {{- end }}
>           {{- end }}
> ```

Thanks for the nice suggestion. I applied it!

File: charts/netdata/templates/daemonset.yaml [hunk 1]
```diff
@@ -95,6 +95,18 @@ spec:
...
+          {{- with .Values.child.claiming }}
+          {{- if and .enabled .token .rooms }}
+          command:
+            - "/bin/sh"
+            - "-c"
+            - >-
+              exec /usr/sbin/run.sh
+                -W set2 cloud global enabled true
+                -W set2 cloud global "cloud base url" "{{ .url }}"
+                -W "claim -token={{ .token }} -rooms={{ .rooms }} -url={{ .url }}"
```
Comment (ilyam8) [inline charts/netdata/templates/daemonset.yaml:107]: Does it work without backslashes?

**Review by Ferroin — DISMISSED** [review]

File: charts/netdata/templates/daemonset.yaml (hunk 1 above)
Comment (knatsakis) [reply charts/netdata/templates/daemonset.yaml]: Actually no. Thanks for spotting this! 🙇🏻 

It works if you remove the indentation. I have pushed it (tested it and it works)

**Review by Ferroin — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

**Pull request merged**